src/anm/analytics.js

```diff
@@ -0,0 +1,76 @@
+var engine = require('engine');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = supportSendBeacon ? 2000 : 1000,
```

Comment:
I would like not to use timeout at all for sendBeacon, and invoke the function only when tab is closing, but I'm afraid that it will not always work.